src/cdk/v2/destinations/intercom/utils.js

```diff
@@ -369,6 +371,111 @@ const addMetadataToPayload = (payload) => {
   return finalPayload;
 };
 
+/**
+ * Api call to attach user to the company
+ * Ref doc v1: https://developers.intercom.com/docs/references/1.4/rest-api/users/companies-and-users/
+ * Ref doc v2: https://developers.intercom.com/docs/references/2.10/rest-api/api.intercom.io/Contacts/attachContactToACompany/
+ * @param {*} payload
+ * @param {*} endpoint
+ * @param {*} destination
+ */
+const attachContactToCompany = async (payload, endpoint, destination) => {
+  let { apiVersion } = destination.Config;
+  apiVersion = isDefinedAndNotNull(apiVersion) ? apiVersion : 'v2';
+  let endpointPath = '/contact/{id}/companies';
+  if (apiVersion === 'v1') {
+    endpointPath = '/users';
+  }
+  const headers = getHeaders(destination, apiVersion);
+  const finalPayload = JSON.stringify(removeUndefinedAndNullValues(payload));
+  const response = await httpPOST(
+    endpoint,
+    finalPayload,
+    {
+      headers,
+    },
+    {
+      destType: 'intercom',
+      feature: 'transformation',
+      endpointPath,
+      requestMethod: 'POST',
+      module: 'router',
+    },
+  );
+
+  const processedResponse = processAxiosResponse(response);
+  if (!isHttpStatusSuccess(processedResponse.status)) {
+    throw new NetworkError(
+      `Unable to attach Contact or User to Company due to : ${JSON.stringify(
+        processedResponse?.response?.errors,
+      )}`,
+      processedResponse?.status,
+      {
+        [tags]: getDynamicErrorType(processedResponse?.status),
+      },
+      processedResponse,
+    );
+  }
+};
+
+/**
+ * Api calls to add or update tags to the company
+ * Ref doc v1: https://developers.intercom.com/docs/references/1.4/rest-api/tags/tag-or-untag-users-companies-leads-contacts/
+ * Ref doc v2: https://developers.intercom.com/docs/references/2.10/rest-api/api.intercom.io/Tags/createTag/
+ * @param message
+ * @param destination
+ * @param id
+ * @returns
+ */
+const addOrUpdateTagsToCompany = async (message, destination, id) => {
+  const companyTags = message?.context?.traits?.tags;
+  if (!companyTags) return;
+  const { apiVersion } = destination.Config;
+  const headers = getHeaders(destination, apiVersion);
+  const baseEndPoint = getBaseEndpoint(destination);
+  const endpoint = `${baseEndPoint}/${TAGS_ENDPOINT}`;
+  const statTags = {
+    destType: 'intercom',
+    feature: 'transformation',
+    endpointPath: '/tags',
+    requestMethod: 'POST',
+    module: 'router',
+  };
+  await Promise.all(
+    companyTags.map(async (tag) => {
+      const finalPayload = {
+        name: tag,
+        companies: [
+          {
+            id,
+          },
+        ],
+      };
+      const response = await httpPOST(
+        endpoint,
+        finalPayload,
+        {
+          headers,
+        },
+        statTags,
+      );
+      const processedResponse = processAxiosResponse(response);
+      if (!isHttpStatusSuccess(processedResponse.status)) {
+        throw new NetworkError(
```

Comment:
Every time it should not be network error, if fails for faulty authorization, it is a configuration error,  if company not found, then instrumentation error. Lets take a second look it response parsing and throw error accordingly.

Reply:
made some changes. please check if it looks good.